### ISSUE TITLE
NVDA leaves a phantom icon in the systray the first time it's restarted after a system boot

### ISSUE THREAD

**Issue by XLTechie** [nvaccess/nvda#11553]
### Steps to reproduce:

1.  Restart Windows.
2.  If NVDA doesn't start automatically after sign-on, start it.
3.  Restart it with the desktop shortcut.
4.  Press Windows+B to go to the notification area.
5.  Use the down arrow to find the NVDA icon.
6.  Press down arrow again.

### Actual behavior:

You should hear a second NVDA icon in the notification area.

In earlier versions of NVDA, I've had as many as four that I can remember.

Pressing enter on the phantom NVDAs will make them go away; only one, usually the lower most in down arrow navigation mode, is the real one.

### Expected behavior:

Only ever one NVDA icon.

### History:

I have experienced this for a number of years.  Lately it only seems to happen after the first run following a system restart; in the past it would happen more often.

This [was recently described by a few users](https://nvda.groups.io/g/nvda/topic/76317611) on a mailing list, and @Qchristensen asked for logs. Find them below.

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

alpha-20809,f4cb0847

#### Windows version:

10 version 2004, but I have experienced this since Windows Vista.

#### Does the issue still occur after restarting your computer?

Yes, these days only after restarting.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Yes, all of them. The same or worse.

#### If addons are disabled, is your problem still occuring?

I'm pretty sure I have tried that in the past, but since it's inconvenient to restart the system that way, I did not try it this time.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?

In this specific case no, but I run it a couple times per week anyway and it has never made a difference, on this computer or any other.

### Logs:

I have included two logs.
One from the version of NVDA which was running first (on system startup), and one from the NVDA which ran second, and actually showed the two icons.

These logs are very long. But reading through them as quickly as I could, the only obvious error I found during the shutdown of the first NVDA was:

```
DEBUG - core._terminate (17:09:21.874) - MainThread (10124):
Terminating UIA support
DEBUGWARNING - _UIAHandler.UIAHandler.terminate (17:09:22.085) - MainThread (10124):
Timeout or error while waiting for UIAHandler MTA thread
```

I have no idea if that is indicative of the problem, or if that happens every time this issue occurs. Nor do I have any idea what might have caused that.

Here are the logs:
* [NVDA-phantom NVDA icon in systray following restart (before).txt](https://github.com/nvaccess/nvda/files/5172685/NVDA-phantom.NVDA.icon.in.systray.following.restart.before.txt)
* [NVDA-phantom NVDA icon in systray following restart (after).txt](https://github.com/nvaccess/nvda/files/5172687/NVDA-phantom.NVDA.icon.in.systray.following.restart.after.txt)

**Comment by XLTechie** [comment]
I want to say that this problem is frustratingly illusive.

I have just attempted several times to provoke it with add-ons disabled in 
either or both of the NVDA runs associated with the behavior, and could not.

Then I tried again to reproduce it according to the steps I used above, and 
could not.

The fact is that this either happens or it doesn't. It is completely unclear to 
me what causes or prevents it, and even I can't replicate it on demand.

I apologize for submitting such a nebulas ticket.

**Comment by ABuffEr** [comment]
Hi,
have you tried killing NVDA from task manager?

**Comment by XLTechie** [comment]
Alberto Buffolino wrote:

> have you tried killing NVDA from task manager?

No.  When would doing that make sense?

During the first run, nothing is apparently wrong.

During the second run, the double icon appears from the first run not fully 
closing.

So by the second run it's too late, and in the first run we have no way of 
knowing whether it's going to happen this time around or not.

I could of course just kill every first run NVDA with taskkill instead of 
desktop shortcut to restart, but that is not what the normal user is going to 
do.

**Comment by ABuffEr** [comment]
> I could of course just kill every first run NVDA with taskkill instead of
desktop shortcut to restart, but that is not what the normal user is going to
do.

Absolutely. I suggested killing only to simulate a not fully closing.

For curiosity, you have Windows Fast Startup enabled?

**Comment by Brian1Gaff** [comment]
Interestingly, yes this goes back to the xp version, but its not the only 
software to have the issue. I've encountered it with things like unchecky 
and other small programs that run at start up but when closed the icon 
remains unless you click it, and suddenly, Windows realises its not running 
any more.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by mohdshara** [comment]
I too experience this behavior. unsure how I can help in this regard. I'll try to watch for a reliable way to reproduce it.

**Comment by Adriani90** [comment]
Yes i see this behavior as well.

cc: @JulienCochuyt do you have any suggestions?

**Comment by ABuffEr** [comment]
Hi,
I know it's not a solution to the issue, that should be investigated and solved. But, if you want to mitigate the phantom icon for all programs (not only NVDA), try [this very micro software.](https://www.novirusthanks.org/products/systray-refresh/)

**Comment by seanbudd** [comment]
Can you confirm if #12286 (now on alpha) fixes this?

**Comment by CyrilleB79** [comment]
Just to inform that this issue is not totally resolved. I have had it just today with NVDA 2021.2beta1.
It seems to occur less frequently than in the past however.

**Comment by Adriani90** [comment]
@XLTechie, @CyrilleB79 are you still seeing this issue in NVDA 2024.3 Beta?

**Comment by CyrilleB79** [comment]
I have not tested the specific STR of this issue.

I have installed NVDA 2024.3beta1 two days ago and may not have the issue of the phantom icon until now. But surely, I have had it with NVDA 2024.2.

**Comment by XLTechie** [comment]
@Adriani90
I have run into this issue as recently as last month. Unfortunately, I don't recall which version I was running then, or on which system.

Are you aware of a recent PR that should have fixed this behavior?

**Comment by Adriani90** [comment]
Yes there were multiple PRs that kind of tried to address terminating NVDA properly.
See e.g. #16934, #16220, etc.

@XLTechie are you still having this issue with NVDA 2025.1 Beta?
